moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -127,36 +127,54 @@ def add_plane(self, name, pose, normal=(0, 0, 1), offset=0):
         co = self.make_plane(name, pose, normal, offset)
         self.__submit(co, attach=False)
 
-    def attach_object(self, attached_collision_object):
-        """Attach an object in the planning scene"""
-        self.__submit(attached_collision_object, attach=True)
+    def attach_object(self, object, link, touch_links=None):
+        """Attach an object to the given link"""
+        if isinstance(object, str):
+            object = self.__make_existing(object)
+        if isinstance(object, CollisionObject):
+            object = AttachedCollisionObject(object=object)
+
+        object.link_name = link
+        if touch_links is not None:
+            object.touch_links = touch_links if touch_links is not None else [link]
+
+        self.__submit(object, attach=True)
```

Comment:
Why not get rid of the attach parameter here altogether? The current code will break with an `AttachedCollisionObject` and `attach == False` and vice versa, right? So we might as well check via `isinstance` as you do above.

Reply:
:+1:

Reply:
The attach argument is needed for __submit to actually add an attached object. There is no type-checking in that function.

Reply:
Yes, there isn't. I rather suggested to add it there. I see why it wasn't there originally in the "never check types" spirit, but then providing the type as a boolean argument isn't much better...

Reply:
Please feel free to augment my PR according to your idea. I still don't get it in full detail and I cannot assess the consequences of this change right now. At least, it is an API change - but of a private function. So, that shouldn't pose a big problem.
Sorry, I'm still somewhat dizzy due to my illness...